src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -120,7 +126,16 @@
             throw new IOException("Unsupported credential type: " + credentials.getClass().getName());
         }
 
-        return gb.build();
+        GitHub github = gb.build();
+        try {
+            github.checkApiUrlValidity();
```

Comment:
Well the output could be prettified in various ways but it is clear enough as is. Not worth adding an extra round-trip to _normal_ reindexing IMO.